src/PlotDock.cpp

```diff
@@ -119,16 +133,32 @@ void PlotDock::updatePlot(SqliteTableModel* model, BrowseDataTableSettings* sett
             for(int i=0;i<model->columnCount();++i)
             {
                 QVariant::Type columntype = guessDataType(model, i);
-                if(columntype != QVariant::String && columntype != QVariant::Invalid)
+                if(columntype != QVariant::Invalid)
                 {
                     QTreeWidgetItem* columnitem = new QTreeWidgetItem(ui->treePlotColumns);
-                    // maybe i make this more complicated than i should
-                    // but store the model column index in the first 16 bit and the type
-                    // in the other 16 bits
-                    uint itemdata = 0;
-                    itemdata = i << 16;
-                    itemdata |= columntype;
-                    columnitem->setData(PlotColumnField, Qt::UserRole, itemdata);
+
+                    switch (columntype) {
+                    case QVariant::DateTime:
+                        columnitem->setText(PlotColumnType, "Date/Time");
+                        break;
+                    case QVariant::Date:
+                        columnitem->setText(PlotColumnType, "Date");
+                        break;
+                    case QVariant::Time:
+                        columnitem->setText(PlotColumnType, "Time");
+                        break;
+                    case QVariant::Double:
+                        columnitem->setText(PlotColumnType, "Numeric");
+                        break;
+                    case QVariant::String:
+                        columnitem->setText(PlotColumnType, "Label");
+                        break;
```

Comment:
Can you add a ```default``` branch? Without one it's producing tons of warnings for me :wink:

Reply:
Fixed. I don't get any warnings, either compiler version difference or difference in flags.